### ISSUE TITLE
Exposed interface of encode/decode

### ISSUE THREAD

**Issue by noctux** [sysread/TOML-Tiny#2]
Hello,

thank you for your library and supporting TOMLv5 in it.
However, I noticed a small discrepancy in the exported public interface of encode, decode and parse (I think):

```pod
=head2 decode

Decodes C<TOML> and returns a hash ref. Dies on parse error.

=head2 encode

Encodes a perl hash ref as a C<TOML>-formatted string. Dies when encountering
an array of mixed types if C<strict_arrays> was set.

=head2 parse

Alias for C<encode> to provide compatibility with C<TOML::Parser> when
overriding the parser by setting C<$TOML::Parser>.
```

So, I would assume that the job of decode is to take a string containing toml-data and return a perl datastructure. The job of encode on the other hand would be to take a perl datastructure and return a string containing the toml representation. However, this does not align well with the documentation for parse (which again  supposedly does toml->perl) but is advertised as an alias to encode.

The code, on the other hand does `write_toml` on `decode` and a (potentially wrong, as it is missing the argument(?)) call to `Parser::parse` on `encode`, which looks wrong to me.

I think the code should look more like that?

```patch
---
 lib/TOML/Tiny.pm | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/lib/TOML/Tiny.pm b/lib/TOML/Tiny.pm
index 8025252..e980490 100644
--- a/lib/TOML/Tiny.pm
+++ b/lib/TOML/Tiny.pm
@@ -43,12 +43,12 @@ sub new {
   bless{ %param, parser => TOML::Tiny::Parser->new(%param) }, $class;
 }
 
-sub encode {
+sub decode {
   my ($self, $source) = @_;
-  $self->{parser}->parse;
+  $self->{parser}->parse($source);
 }
 
-sub decode {
+sub encode {
   my ($self, $data) = @_;
   TOML::Tiny::Writer::to_toml($data,
     strict_arrays => $self->{strict_arrays},
@@ -59,7 +59,7 @@ sub decode {
 # For compatibility with TOML::from_toml's use of $TOML::Parser
 #-------------------------------------------------------------------------------
 sub parse {
-  goto \&encode;
+  goto \&decode;
 }
 
 1;
@@ -237,7 +237,7 @@ an array of mixed types if C<strict_arrays> was set.
 
 =head2 parse
 
-Alias for C<encode> to provide compatibility with C<TOML::Parser> when
+Alias for C<dencode> to provide compatibility with C<TOML::Parser> when
 overriding the parser by setting C<$TOML::Parser>.
 
 
-- 
```

Is my intuition correct, or am I misunderstanding something?
I'd be happy to propose this as a pull request if that was more convenient to you.

Thanks again for your work and reading this issue thus far,
Simon

**Comment by sysread** [comment]
Thanks for catching that. I'll fix the OO interface immediately.

**Comment by sysread** [comment]
This is fixed in v0.06

**Comment by noctux** [comment]
Perfect, thanks for the fast reponse!
